presto-cli/src/main/java/io/prestosql/cli/Console.java

```diff
@@ -309,7 +309,7 @@ private static boolean process(QueryRunner queryRunner, String sql, OutputFormat
         }
 
         try (Query query = queryRunner.startQuery(finalSql)) {
-            boolean success = query.renderOutput(System.out, outputFormat, interactive);
+            boolean success = query.renderOutput(printStream, outputFormat, interactive);
```

Comment:
I don't think this works. The existing code is rather messy and these parameters are used for multiple things:

* `out` is used for
  * query results
  * errors in interactive mode (status and errors go to `stdout` since it is a terminal)
* `interactive` controls
  * if `out` is used for errors (or if errors go to `stderr`)
  * whether or not to show interactive status
  * whether to send the output through a pager (like `less`) or to simply print it

I think we can change this as follows:

*  Change `renderOutput` to take both out `out` and `errorChannel` streams. The `interactive` flag would no longer control it.
* In `renderQueryOutput`, `new PrintStreamWarningsPrinter(...)` should use `errorChannel` rather than hard-coding `System.err`, and should move inside the `else` block (since it isn't used in interactive mode).
* Add a new `progress` flag that is used to decide if progress should be shown.